### DISCUSSION TITLE
5. December 2023

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#1660] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [ ] Dmitrii: discuss #1423 (how to sync files updated on the host)
   - Comment from @mkow: "I agree that we don't really have the semantics of chroot mounts well defined if the host changes the files when the enclave is still running. We should decide on that before merging this PR."
- [x] Woju: draft proposal for an exception for examples

### Dmitrii: discuss #1423

This was postponed to a next meeting, when @mkow will be able to join. Since @mkow left the comment about the semantics of our FS implementation, it doesn't make sense to discuss the implementation without his direct participation.

### Woju: draft proposal for an exception for examples

- Dmitrii and Mona: suggested a small fix to the draft ("if not already licensed under a permissive licence").

After that, @woju started a **formal voting procedure (with 6 voters)**:
- Present maintainers: Dmitrii, Mona, Don, Chia-Che, Woju, Kailun. Not present: Michal, Borys. (See the current maintainer list [here](https://github.com/gramineproject/gramine/blob/fa437cbcf380ea9c1aa76958ddeb73557a344901/Documentation/management-team.rst).)
- All present maintainers voted "yes":
   - 6 people voted "yes",
   - 0 people voted "no",
   - 0 people voted "abstain".
- This is 2/3 of the entire TSC (maintainers), so the proposal is accepted.

See the follow up PR created by @woju : https://github.com/gramineproject/gramine/pull/1670

**Comment by woju** [comment]
- [ ] woju: draft proposal for an exception for examples

```
Gramine Technical Steering Committee,

- acting in accordance with the Technical Charter, adopted on 14 September 2020,

Whereas:

(1) As stated in Mission and Scope of the Project (Charter §1-2), it is one of
    the goals of the project to make adoption of Gramine as wide and as easy as
    possible.

(2) Downstream projects that use, but not modify Gramine, might be licensed
    under terms incompatible with LGPL-3, which is fine. Nevertheless they need
    to copy and modify some boilerplate (like manifests), which might be
    available from our project.

(3) The directory `CI-Examples/` in main project repository, and the `examples`
    auxiliary repository (hereinafter “Examples”) serve as examples, templates
    and/or boilerplate for downstream projects that use gramine. Those Examples
    should be available for reuse by downstream projects under permissive
    licence.

Resolves:

1. That as a special exception, in accordance with §7.3 of the Charter,
   Examples are allowed to be accepted under permissive licence approved by OSI
   (e.g., BSD-3-Clause or another).

   What constitutes “example code” that fits Examples is left to the discretion
   of the reviewers who accept the relevant contribution.

2. That the project is to attempt to relicense existing Examples by asking
   authors and/or copyright holders thereof to relicense their respective
   contributions under BSD-3-Clause licence.
```

I intend to give an opportunity for all people who attend (even non-voting) to propse amendments. Then after we'll have consensus on the text, we'll either vote if we'll have quorum, or if we won't, we'll vote asynchronously on some private channel and submit results in another meeting.

**Reply by dimakuv** [reply]
> I intend to give an opportunity for all people who attend (even non-voting) to propse amendments. 

We have C code that was taken from mbedTLS examples and modified: https://github.com/gramineproject/gramine/tree/master/CI-Examples/ra-tls-mbedtls/src. This code is licensed by mbedTLS as Apache 2.0, so we won't be able to change these files to the BSD license. I don't know if this requires any amendment in your proposed text; just wanted to highlight this case.

The rest looks good to me. All other files in those dirs/repos can be modified.

**Reply by woju** [reply]
Yeah, good point. This is the kind of comment that I'd like to hear.
